tools/ci_build/github/apple/build_ios_framework.py

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import argparse
+import glob
+import json
+import os
+import pathlib
+import shutil
+import subprocess
+import sys
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+REPO_DIR = os.path.normpath(os.path.join(SCRIPT_DIR, "..", "..", "..", ".."))
+BUILD_PY = os.path.join(REPO_DIR, "tools", "ci_build", "build.py")
+
+# We by default will build below 2 archs
+DEFAULT_BUILD_OSX_ARCHS = [
+    {'sysroot': 'iphoneos', 'arch': 'arm64'},
+    {'sysroot': 'iphonesimulator', 'arch': 'x86_64'},
+]
+
+
+def _parse_build_settings(args):
+    _setting_file = args.build_settings_file.resolve()
+
+    if not _setting_file.is_file():
+        raise FileNotFoundError('Build config file {} is not a file.'.format(_setting_file))
+
+    with open(_setting_file) as f:
+        _build_settings_data = json.load(f)
+
+    build_settings = {}
+
+    if 'apple_deploy_target' in _build_settings_data:
+        build_settings['apple_deploy_target'] = _build_settings_data['apple_deploy_target']
+    else:
+        raise ValueError('apple_deploy_target is required in the build config file')
+
+    build_settings["build_osx_archs"] = _build_settings_data.get("build_osx_archs", DEFAULT_BUILD_OSX_ARCHS)
+
+    build_params = []
+    if 'build_params' in _build_settings_data:
+        build_params += _build_settings_data['build_params']
+    else:
+        raise ValueError('build_params is required in the build config file')
+
+    build_settings['build_params'] = build_params
+    return build_settings
+
+
+def _build_package(args):
+    build_settings = _parse_build_settings(args)
+    build_dir = os.path.abspath(args.build_dir)
+
+    # Temp dirs to hold building results
+    _intermediates_dir = os.path.join(build_dir, 'intermediates')
+    _build_config = args.config
+    _base_build_command = [
+        sys.executable, BUILD_PY,
+        '--config=' + _build_config,
```

Comment:
these options could be specified again in build_settings['build_params'] and overridden. if we want these to take precedence, put them at the end

Reply:
These 2 I would prefer them to be separated from build_params
For example, you may want to build a Debug flavor without changing any file.

Reply:
so if they happen to be in build_params as well, those values will take precedence. is that ok?

Reply:
I can move apple_deploy_target to the build_params 
However I would still prefer --config to be separated, if we use xcode to build ort, the actual binary folder is <build folder>/<_build_config>-<_sysroot>, it is easier to have it as a param instead of parsing the build_settings['build_params']